infra/cifuzz/run_fuzzers.py

```diff
@@ -202,6 +209,55 @@ class BatchFuzzTargetRunner(BaseFuzzTargetRunner):
   def quit_on_bug_found(self):
     return False
 
+  def run_fuzz_target(self, fuzz_target_obj):
+    """Fuzzes with |fuzz_target_obj| and returns the result."""
+    result = fuzz_target_obj.fuzz()
+    logging.debug('corpus_path: %s', os.listdir(result.corpus_path))
+    self.clusterfuzz_deployment.upload_corpus(fuzz_target_obj.target_name,
+                                              result.corpus_path)
+    # This must be done after we upload the corpus, otherwise it will be deleted
+    # before we get a chance to upload it. We can't delete the fuzz target
+    # because it is needed when we upload the build.
+    fuzz_target_obj.free_disk_if_needed(delete_fuzz_target=False)
+    return result
+
+  def run_fuzz_targets(self):
+    result = super().run_fuzz_targets()
+
+    self.clusterfuzz_deployment.upload_crashes(self.crashes_dir)
+
+    # We want to upload the build to the filestore after we do batch fuzzing.
+    # There are some problems with this. First, we don't want to upload the
+    # build before fuzzing, because if we download the latest build, we will
+    # consider the build we just uploaded to be the latest even though it
+    # shouldn't be (we really intend to download the build before the curent
+    # one.
+    # Second, the out directory is mounted into the runnner container and is
+    # used to pass the runner corpora, old builds and for the runner to pass the
+    # host testcases. Thus, we will upload the build after fuzzing. But before
+    # we zip the out directory we will remove these extra things that are now in
```

Comment:
this does seem very brittle. we should definitely rethink how we can do refactor things to do this in a more clean way.

Reply:
For sure. I'd like to change this /out issue, but there is something that prevents me from doing so.
Most repos' workflow files upload artifacts from a hardcoded directory out/artifacts. We will need to migrate these users to stop abusing /out.